internal/services/datafactory/validate/ssh_private_key.go

```diff
@@ -0,0 +1,21 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func SSHPrivateKey(i interface{}, k string) (warnings []string, errors []error) {
+	value, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected %q to be a string", k))
+		return
+	}
+	pattern := `(?s)-----BEGIN (RSA|DSA) PRIVATE KEY-----.*?-----END (RSA|DSA) PRIVATE KEY-----`
+
+	if !regexp.MustCompile(pattern).MatchString(value) {
+		errors = append(errors, fmt.Errorf("invalid private key, must be in OpenSSH PEM format %q: %q", k, value))
```

Comment:
Since we've marked the key as sensitive in the schema we probably shouldn't log it in the error message
```suggestion
		errors = append(errors, fmt.Errorf("invalid private key, %q must be in OpenSSH PEM format", k))
```

Reply:
oh good catch, thanks!